### PR TITLE
Change override flow

### PR DESCRIPTION
**Pull request by eyalpalo**
Changed override flow to override corepacks.json files only if the file corepacks-override.json has changed since last upload commit.

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-9292)

### CODE REVIEW

File: Tests/Marketplace/override_core_packs_versions.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    last_commit_blob = commit.tree / file_path
+
+    # Get the content of the last commit blob
+    last_commit_content = json.loads(last_commit_blob.data_stream.read())
+
+    # Get the current file content
+    current_override_content = GCPConfig.corepacks_override_contents
+
+    # If the files are different and the current marketplace is in the override file we override
+    if current_override_content != last_commit_content and marketplace in current_override_content:
```
Comment (dantavori) [inline Tests/Marketplace/override_core_packs_versions.py:42]: shouldn't it be:
```suggestion
    if current_override_content != last_commit_content and marketplace in current_override_content.get("marketplaces", []):
```

**Review by dantavori — APPROVED** [review]
overall looks good, see notes
before merging:
1. run test-upload on this branch with relevant test cases, most importantly updating corepacks_override.json with multiple marketplaces and ensuring the files are updated in the "prod" bucket as expected
2. keep the `corepacks_override.json` updated with the new schema (but `updated_corepacks_content` field empty, so it will not trigger an override in production)

File: Tests/Marketplace/override_core_packs_versions.py (hunk 1 above)
Comment (eyalpalo) [reply Tests/Marketplace/override_core_packs_versions.py:42]: no, the new formation is as follows:
server_verions: x_x_x
xsoar_saas: {
      updated_corepacks_content: {...}
}

File: Tests/Marketplace/override_core_packs_versions.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+        Args
+            marketplace (str): the marketplace type of the bucket. possible options: xsoar, xsoar_saas, marketplace_v2 or xpanse
+            last_upload_commit(str): Last upload commit.
+
+        Returns True if a file should be updated and False otherwise.
+        """
+
+    content_repo = get_content_git_client(CONTENT_ROOT_PATH)
+    logging.info(last_upload_commit)
```
Comment (yaakovpraisler) [inline Tests/Marketplace/override_core_packs_versions.py:27]: remove

**Review by yaakovpraisler — COMMENTED** [review]
Looks good to me. Good job!

**Comment by eyalpalo** [comment]
https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/869081 nightly run

**Pull request merged**